packages/simcore-sdk/src/simcore_sdk/node_ports_common/r_clone.py

```diff
@@ -0,0 +1,134 @@
+import asyncio
+import logging
+import re
+from contextlib import asynccontextmanager
+from pathlib import Path
+from typing import AsyncGenerator, Optional
+
+from aiocache import cached
+from aiofiles import tempfile
+from aiohttp import ClientSession
+from models_library.users import UserID
+from pydantic.errors import PydanticErrorMixin
+from settings_library.r_clone import RCloneSettings
+from settings_library.utils_r_clone import get_r_clone_config
+
+from .constants import ETag
+from .storage_client import (
+    delete_file_meta_data,
+    get_upload_file_presigned_link,
+    update_file_meta_data,
+)
+
+logger = logging.getLogger(__name__)
+
+
+class _CommandFailedException(PydanticErrorMixin, RuntimeError):
+    msg_template: str = "Command {command} finished with exception:\n{stdout}"
+
+
+@asynccontextmanager
+async def _config_file(config: str) -> AsyncGenerator[str, None]:
+    async with tempfile.NamedTemporaryFile("w") as f:
+        await f.write(config)
+        await f.flush()
+        yield f.name
+
+
+async def _async_command(*cmd: str, cwd: Optional[str] = None) -> str:
+    str_cmd = " ".join(cmd)
+    proc = await asyncio.create_subprocess_shell(
+        str_cmd,
+        stdin=asyncio.subprocess.PIPE,
+        stdout=asyncio.subprocess.PIPE,
+        stderr=asyncio.subprocess.STDOUT,
+        cwd=cwd,
+    )
+
+    stdout, _ = await proc.communicate()
+    decoded_stdout = stdout.decode()
+    if proc.returncode != 0:
+        raise _CommandFailedException(command=str_cmd, stdout=decoded_stdout)
+
+    logger.debug("'%s' result:\n%s", str_cmd, decoded_stdout)
+    return decoded_stdout
+
+
+@cached()
+async def is_r_clone_available(r_clone_settings: Optional[RCloneSettings]) -> bool:
+    """returns: True if the `rclone` cli is installed and a configuration is provided"""
+    try:
+        await _async_command("rclone", "--version")
+        return r_clone_settings is not None
+    except _CommandFailedException:
+        return False
+
+
+async def sync_local_to_s3(
+    session: ClientSession,
+    r_clone_settings: RCloneSettings,
+    s3_object: str,
+    local_file_path: Path,
+    user_id: UserID,
+) -> ETag:
+
+    s3_link = await get_upload_file_presigned_link(
```

Comment:
Please consider renaming the fct if it does not return a presigned link. something like get_upload_link or something similar.